### PR TITLE
[4.3.4] Remove the virtual mount display id from dead creatures

### PR DESCRIPTION
**Pull request by ghost**
In order to fix that strange messed up visual.
Found this handling while doing some Zul'Gurub on retail.

### CODE REVIEW

**Comment by DDuarte** [comment]
Should this happen on 3.3.5 as well?

**Comment by NorthStrider** [comment]
counts for all branches, ye

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -1482,6 +1482,9 @@ void Creature::setDeathState(DeathState s)
         SetTarget(ObjectGuid::Empty);                // remove target selection in any cases (can be set at aura remove in Unit::setDeathState)
         SetUInt32Value(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_NONE);
 
+        if (GetUInt32Value(UNIT_FIELD_MOUNTDISPLAYID)) // if creature is mounted on a virtual mount, remove it at death
```
Comment (joschiwald) [inline src/server/game/Entities/Creature/Creature.cpp]: no need to check the old value, setter will do it

Comment (Kittnz) [inline src/server/game/Entities/Creature/Creature.cpp]: So remove the if line && merge && cherry pick :+1:

**Pull request closed without merging**